src/utilities/flattenRoutes.ts

```diff
@@ -13,19 +14,26 @@ export function flattenRoutes(routes: Routes, path = ''): RouteFlat[] {
     if (isNamedRoute(route)) {
       value.push({
         name: route.name,
-        regex: generateRouteRegexPattern(fullPath),
         path: fullPath,
+        regex: generateRouteRegexPattern(fullPath),
+        params: getRouteParams(route.path),
       })
     }
 
     return value
   }, [])
 }
 
-export function generateRouteRegexPattern(path: string): RegExp {
+export function getRouteParams(value: string | Path): Record<string, Param[]> {
```

Comment:
This could probably return `ExtractParamsFromPath<T>`

Reply:
moved to `types/routeFlat.ts`